ts/features/services/components/ServiceDetailsPreferences.tsx

```diff
@@ -0,0 +1,144 @@
+    channels,
+    RA.findFirst(channel => channel === notificationChannel),
+    O.isSome
+  );
+
+export type ServiceDetailsPreferencesProps = {
+  serviceId: ServiceId;
+  availableChannels?: ReadonlyArray<NotificationChannelEnum>;
+};
+
+export const ServiceDetailsPreferences = ({
+  serviceId,
+  availableChannels = []
+}: ServiceDetailsPreferencesProps) => {
+  const isFirstRender = useFirstEffect();
+
+  const dispatch = useIODispatch();
+
+  const servicePreferenceResponseSuccess = useIOSelector(
+    servicePreferenceResponseSuccessSelector
+  );
+
+  const isErrorServicePreference = useIOSelector(
+    isErrorServicePreferenceSelector
+  );
+
+  const isPremiumMessagesOptInOutEnabled = useIOSelector(
+    isPremiumMessagesOptInOutEnabledSelector
+  );
+
+  const serviceMetadataInfo = useIOSelector(state =>
+    serviceMetadataInfoSelector(state, serviceId)
+  );
+
+  const isInboxPreferenceEnabled = pipe(
+    servicePreferenceResponseSuccess,
+    O.fromNullable,
+    O.map(servicePreference => servicePreference.value.inbox),
+    O.getOrElse(() => false)
+  );
+
+  useEffect(() => {
+    if (!isFirstRender && isErrorServicePreference) {
+      IOToast.error(I18n.t("global.genericError"));
+    }
+  }, [isFirstRender, isErrorServicePreference]);
+
+  const handlePreferenceValueChange = useCallback(
+    (channel: keyof EnabledChannels, value: boolean) => {
+      if (servicePreferenceResponseSuccess) {
+        dispatch(
+          upsertServicePreference.request({
+            id: serviceId,
+            ...servicePreferenceResponseSuccess.value,
+            [channel]: value
+          })
+        );
+      }
+    },
+    [dispatch, serviceId, servicePreferenceResponseSuccess]
+  );
+
+  return (
+    <>
+      <ListItemHeader
+        label={I18n.t("services.details.preferences.title")}
+        accessibilityLabel={I18n.t("services.details.preferences.title")}
+      />
+      {/* 
+        this switch is disabled if the current service is a special service.
+        the user can enable the service only using the proper special service flow.
+      */}
+      <ServicePreferenceListItemSwitch
+        channel="inbox"
+        icon="message"
+        disabled={serviceMetadataInfo?.isSpecialService}
+        label={I18n.t("services.details.preferences.inbox")}
+        onPreferenceValueChange={(value: boolean) =>
+          handlePreferenceValueChange("inbox", value)
+        }
+      />
+
+      {isInboxPreferenceEnabled ? (
+        <>
+          {hasChannel(NotificationChannelEnum.WEBHOOK, availableChannels) && (
+            <>
+              <Divider />
+              <ServicePreferenceListItemSwitch
+                channel="push"
+                icon="bell"
+                label={I18n.t("services.details.preferences.pushNotifications")}
+                onPreferenceValueChange={(value: boolean) =>
+                  handlePreferenceValueChange("push", value)
+                }
+              />
+            </>
+          )}
```

Comment:
Why not use a FlatList to render these items, including the condition? Using `FlatList` you can easily manage `Divider` components in a more maintainable way (if you forget to add a `Divider` manually, you break the layout design). Also, you don't have to create a new component that doesn't add that much value compared to `ListItemSwitch` itself. What do you think?

(Check out the Developer Mode component to see some examples)

Reply:
Done it https://github.com/pagopa/io-app/pull/5658/commits/85675797b9be4d9549cd9972aad8ea4c2f6f7d29